### ISSUE TITLE
Stop recommending package:http

### ISSUE THREAD

**Issue by matanlurey** [dart-lang/site-webdev#1434]
... this package (`package:http`) is too complex/bloated for typical web application usage. When we removed it from our [Hacker News PWA](https://dart-hn.firebaseapp.com/) it shrunk the production size from 346Kb to 274Kb (~21% drop):

https://github.com/dart-lang/angular/commit/9802811fe3c6f211a424d98870538a1ef0dea745#diff-8deb42d79272db3c22d42fb7b9963918

Most users will be fine with the built-in `HttpRequest` class. I don't think we need to go into more details for advanced users on these pages or tutorials.

**Comment by nex3** [comment]
I disagree with the characterization of `http` as "bloated", but I agree that it shouldn't be the default HTTP client recommended for use in web applications. Its goal has always been to provide a cross-platform, composable client API. If you only care about making HTTP requests on the web and you don't care about adding additional middleware on top of the default client, it doesn't make sense to use a wrapper.

**Comment by matanlurey** [comment]
🤷‍♂️ for a web application a ~70Kb HTTP library is a complete no-no.

Maybe if the entire web application was centered around testing HTTP, but there are full websites / applications that are less than 70Kb. I just cannot in good conscious recommend it to _any_ web user of Dart.

**Comment by nex3** [comment]
To be clear, I'd love to bring that number down, but a lot of it is outside the scope of the package itself. Almost half of that is from the core library's `Uri` class, which will hopefully be reduced once the changes in https://github.com/dart-lang/sdk/issues/29420#issuecomment-339263063 are implemented. This also means that if any other code in the app touches the core `Uri` class in any way, the *relative* overhead of using `http` is much smaller.

There are also outstanding tree-shaking issues in dart2js (https://github.com/dart-lang/sdk/issues/30962 and https://github.com/dart-lang/sdk/issues/30963) which would get rid of somewhere in the neighborhood of 16kB. The segment of `http`'s dependencies that are *actually used* in a web app isn't very large, but Dart isn't currently sophisticated enough to recognize that in a way that translates to lower code size.

**Comment by matanlurey** [comment]
Sure, not ascribing fault, just the current state of Dart + tooling + package:http = bad for users.
